data_curation/README.md

```diff
@@ -1,4 +1,350 @@
-# DIBBS Text to Code data_curation
+# DIBBS Text to Code (TTC) - Data Curation
 
-The `data_curation` folder contains scripts for TTC model development, tuning, and
-evaluation.
+## Table of Contents
+
+- [Overview](#overview)
+- [Scripts](#scripts)
+- [Data Files](#data-files)
+   - [LOINC](#loinc)
+   - [LOINC Part Synonyms & Abbreviations](#loinc-part-synonyms-&-abbreviations)
+   - [LOINC Part Descriptions](#loinc-part-descriptions)
+   - [LOINC UMLS Related Names](#loinc-umls-related-names)
+   - [SNOMED](#snomed)
+   - [HL7](#hl7)
+- [Instructions](#instructions)
+   - [Generating SNOINC Extracts](#generating-snoinc-extracts)
+     - [Dependencies](#dependencies)
+     - [Command Line](#command-line)
+     - [Direct Relma DB Queries](#direct-relma-db-queries)
+
+## Overview
+
+The `data_curation` folder contains scripts for TTC model development, tuning, and evaluation.  Most of the scripts leverage data that is being pulled from the LOINC, UMLS, and HL7 APIs.  However, some require the LOINC RelmaDB (MS-Access database).
+
+## Scripts
+
+### terminology_valueset_sync.py
+
+Contains various functions to pull data from SNOMED, LOINC, and HL7 APIs to provide data for the TTC model. For detailed instructions on how to use the various scripts [see instruction section below](#command-line)
+
+### augmentation.py
+
+A gathering of various functions to modify data from the different terminology data sets to help with model training and tuning.  Functions that randomly scramble words, randomly delete characters, and randomly replacing words with related terms are included.  The `configs.py` is leveraged to allow the user to configure the various parameters/properties for these random data modification functions.
+
+### configs.py
+
+A collection of various configurations used to augment data and/or create synthetic data, using the terminology data sets.
+
+### generation.py
+
+A script to house functions that can be used to generate data sets used to help train and tune the data models.  ie. `Generate Positive Pairs` - Given the location of one or more files of LOINC codes and some corresponding augmented examples for those codes, this function compiles a list of positive pairs that can be read for model training. A positive pair is a tuple of the form (original_loinc_code, augmented_example_of_code).
+
+### synthetic_lab_results.py
+
+Generate a CSV of synthetic lab results with labeled values. Each row contains a randomized result word (e.g., "positive", "not detected")
+and a label: 1 for positive terms, 2 for negative terms. Optionally, the cript can introduce randomized case changes and typos.
+
+### loinc (folder)
+
+Contains .sql queries/files that are used to gather data from LOINC's RELMA database (MS-Access), as well as the resulting data files that are used to generate some of the data files.
+
+
+## Data Files
```

Comment:
Do we need all of these data files? It's duplicate of what's in the `data` README

Reply:
I knew this would be a question.  I figured maybe we have the details in one or the other, but also didn't want to leave the information off the actual folder that contains the data.  Maybe just a link to the other readme that contains the details?